File: python/creational/singleton.py
class Singleton:
    __instance = None

    class __Singleton:
        def getUniqueId(self):
            return str(self) + ',id='+str(id(self))

    def __new__(self):
        if Singleton.__instance is None:
             Singleton.__instance = Singleton.__Singleton()
        return Singleton.__instance

s = Singleton() 
print(s)

s1 = Singleton()
print(s1.getUniqueId())

s2 = Singleton()
print(s2.getUniqueId())

File: java/src/patterns/Singleton.java
package patterns;

public class Singleton {
    public static Singleton Instance;

    private Singleton() {

    }

    public static synchronized Singleton getInstance() {
        if(Instance == null) {
            Instance = new Singleton();
        }
        return Instance;
    }

    @Override
    public int hashCode() {
        return 12345;
    }
}
